ql/src/test/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/TestHiveMetaStoreAuthorizer.java

```diff
@@ -547,76 +642,29 @@ public void testUnAuthorizedCause() {
-    otherTableMeta.setOwnerType(org.apache.hadoop.hive.metastore.api.PrincipalType.USER);
-    tableMetas.add(otherTableMeta);
-
-    TableFilterContext filterContext = TableFilterContext.createFromTableMetas(default_db, tableMetas);
-    List<Table> tables = filterContext.getTables();
-    assertEquals("Should have two tables in context", 2, tables.size());
-
-    boolean foundOwnerTable = false;
-    boolean foundOtherTable = false;
-
-    for (Table table : tables) {
-      if (table.getTableName().equals("owner_table")) {
-        foundOwnerTable = true;
-        assertEquals("owner_table should have authorized user as owner", authorizedUser, table.getOwner());
-        assertEquals("owner_table should have correct owner type",
-            org.apache.hadoop.hive.metastore.api.PrincipalType.USER, table.getOwnerType());
-      } else if (table.getTableName().equals("other_table")) {
-        foundOtherTable = true;
-        assertEquals("other_table should have unauthorized user as owner", unAuthorizedUser, table.getOwner());
-        assertEquals("other_table should have correct owner type",
-            org.apache.hadoop.hive.metastore.api.PrincipalType.USER, table.getOwnerType());
+      // Check if the exception chain contains HiveAuthzPluginException
+      Throwable current = e;
+      boolean foundAuthzException = false;
+      while (current != null) {
+        if (current instanceof HiveAuthzPluginException) {
+          foundAuthzException = true;
+          String expectedErrMsg = "Operation type " + HiveOperationType.CREATEDATABASE + " not allowed for user:" + unAuthorizedUser;
+          assertTrue("Expected error message mismatch. Actual: '" + current.getMessage() + "'", current.getMessage().contains(expectedErrMsg));
+          break;
+        }
+        current = current.getCause();
       }
-    }
-
-    assertTrue("owner_table not found in tables", foundOwnerTable);
-    assertTrue("other_table not found in tables", foundOtherTable);
-
-    HiveMetaStoreAuthzInfo authzInfo = filterContext.getAuthzContext();
-    List<HivePrivilegeObject> privObjects = authzInfo.getInputHObjs();
 
-    assertEquals("Should have two privilege objects", 2, privObjects.size());
-
-    foundOwnerTable = false;
-    foundOtherTable = false;
-
-    for (HivePrivilegeObject obj : privObjects) {
-      if (obj.getObjectName().equals("owner_table")) {
-        foundOwnerTable = true;
-        assertEquals("owner_table privilege object should have authorized user as owner",
-            authorizedUser, obj.getOwnerName());
-      } else if (obj.getObjectName().equals("other_table")) {
-        foundOtherTable = true;
-        assertEquals("other_table privilege object should have unauthorized user as owner",
-            unAuthorizedUser, obj.getOwnerName());
+      if (!foundAuthzException) {
+        Throwable rootCause = ExceptionUtils.getRootCause(e);
+        if (rootCause instanceof HiveAuthzPluginException) {
```

Comment:
if `foundAuthzException` is false, then the `rootCause` shouldn't be the `HiveAuthzPluginException`?